File: packages/shared-types/src/enums.ts
// Mirrors backend/app/db/enums.py. Keep the two in sync when either changes.

export const Priority = {
  Critical: "critical",
  High: "high",
  Medium: "medium",
  Low: "low",
  Noise: "noise",
} as const;
export type Priority = (typeof Priority)[keyof typeof Priority];

export const CommitmentOwner = {
  User: "user",
  Counterparty: "counterparty",
} as const;
export type CommitmentOwner =
  (typeof CommitmentOwner)[keyof typeof CommitmentOwner];

export const CommitmentStatus = {
  Open: "open",
  Done: "done",
  Snoozed: "snoozed",
  Dismissed: "dismissed",
} as const;
export type CommitmentStatus =
  (typeof CommitmentStatus)[keyof typeof CommitmentStatus];

export const TaskStatus = {
  Open: "open",
  Done: "done",
  Snoozed: "snoozed",
} as const;
export type TaskStatus = (typeof TaskStatus)[keyof typeof TaskStatus];

export const SourceType = {
  Gmail: "gmail",
  Calendar: "calendar",
  Manual: "manual",
  Voice: "voice",
} as const;
export type SourceType = (typeof SourceType)[keyof typeof SourceType];

export const MessageClassification = {
  NeedsReply: "needs_reply",
  NeedsDecision: "needs_decision",
  Deadline: "deadline",
  MeetingScheduling: "meeting_scheduling",
  FollowUpNeeded: "follow_up_needed",
  WaitingForResponse: "waiting_for_response",
  Informational: "informational",
  LowPriority: "low_priority",
  SpamNoise: "spam_noise",
  Sensitive: "sensitive",
} as const;
export type MessageClassification =
  (typeof MessageClassification)[keyof typeof MessageClassification];

export const ActionStatus = {
  Proposed: "proposed",
  Approved: "approved",
  Rejected: "rejected",
  Executed: "executed",
  Failed: "failed",
} as const;
export type ActionStatus = (typeof ActionStatus)[keyof typeof ActionStatus];

// Action risk levels (PRD 12.10). The slice guards the level-3 boundary.
export const RiskLevel = {
  ReadOnly: 0,
  InternalPrep: 1,
  ReversibleWrite: 2,
  ExternalComm: 3,
  FinancialLegal: 4,
  Sensitive: 5,
} as const;
export type RiskLevel = (typeof RiskLevel)[keyof typeof RiskLevel];

File: backend/app/db/enums.py
"""Enums shared across models and schemas. Mirror packages/shared-types/src/enums.ts."""

import enum


class Provider(enum.StrEnum):
    google = "google"


class SyncStatus(enum.StrEnum):
    never = "never"
    syncing = "syncing"
    ok = "ok"
    error = "error"


class MessageClassification(enum.StrEnum):
    needs_reply = "needs_reply"
    needs_decision = "needs_decision"
    deadline = "deadline"
    meeting_scheduling = "meeting_scheduling"
    follow_up_needed = "follow_up_needed"
    waiting_for_response = "waiting_for_response"
    informational = "informational"
    low_priority = "low_priority"
    spam_noise = "spam_noise"
    sensitive = "sensitive"


class Priority(enum.StrEnum):
    critical = "critical"
    high = "high"
    medium = "medium"
    low = "low"
    noise = "noise"


class CommitmentOwner(enum.StrEnum):
    user = "user"  # the user owes someone something
    counterparty = "counterparty"  # someone owes the user


class CommitmentStatus(enum.StrEnum):
    open = "open"
    done = "done"
    snoozed = "snoozed"
    dismissed = "dismissed"


class TaskStatus(enum.StrEnum):
    open = "open"
    done = "done"
    snoozed = "snoozed"


class SourceType(enum.StrEnum):
    gmail = "gmail"
    sms = "sms"
    calendar = "calendar"
    manual = "manual"
    voice = "voice"


# Action risk levels per PRD section 12.10. Levels 4-5 require strong confirmation.
class RiskLevel(enum.IntEnum):
    read_only = 0
    internal_prep = 1
    reversible_write = 2
    external_comm = 3
    financial_legal = 4
    sensitive = 5


class ActionType(enum.StrEnum):
    create_draft = "create_draft"  # push a reply into Gmail drafts (level 3)
    create_task = "create_task"  # level 2 reversible write
    create_calendar_event = "create_calendar_event"  # level 2: book your own time
    update_calendar_event = "update_calendar_event"  # level 2: reschedule your own time
    delete_calendar_event = "delete_calendar_event"  # level 2: cancel your own time
    send_email = "send_email"  # level 3 external comm
    send_message = "send_message"  # level 3 (e.g. WhatsApp)
    make_payment = "make_payment"  # level 4 financial
    place_order = "place_order"  # level 4 financial (commerce)
    browser_action = "browser_action"  # refused; see docs/integrations/REFUSED.md


class ActionStatus(enum.StrEnum):
    proposed = "proposed"
    approved = "approved"
    rejected = "rejected"
    executed = "executed"
    failed = "failed"


class NotificationType(enum.StrEnum):
    deadline_risk = "deadline_risk"
    meeting_prep = "meeting_prep"
    unanswered_email = "unanswered_email"
    follow_up_due = "follow_up_due"
    daily_briefing = "daily_briefing"
    approval_needed = "approval_needed"
    schedule_conflict = "schedule_conflict"
    reminder = "reminder"
    new_mail = "new_mail"


class NotificationStatus(enum.StrEnum):
    pending = "pending"  # queued, not yet sent (e.g. held by quiet hours)
    sent = "sent"
    suppressed = "suppressed"  # batched away or below threshold


class ScheduleProposalStatus(enum.StrEnum):
    pending = "pending"
    accepted = "accepted"
    dismissed = "dismissed"


# Priority floor a notification type must clear to send immediately vs. batch.
class NotificationImportance(enum.IntEnum):
    low = 1
    normal = 2
    high = 3
